### PR TITLE
Added error definitions and refactored client to use them

### PR DESCRIPTION
**Pull request by Laursen79**

### CODE REVIEW

File: client/src/server.rs [hunk 1]
```diff
@@ -165,57 +136,46 @@ impl Server for ServerAPI {
...
+        self.make_request(ReqType::Put(payload), uri).await;
+        todo!("Handle the response")
     }
 
     async fn send_msg(
         &self,
         msg: String,
         user_id: String,
         device_id: u32,
-    ) -> Result<WebSocketResponseMessage> {
```
Comment (UnluckyBird) [inline client/src/server.rs:176]: Is this not needed anymore?

File: client/src/server.rs [hunk 2]
```diff
@@ -57,82 +52,60 @@ impl Display for ReqType {
...
 
 impl Server for ServerAPI {
+    type Error = SignalClientError;
     async fn connect(
         &mut self,
         username: &str,
         password: &str,
         url: &str,
         port: &str,
-    ) -> Result<()> {
```
Comment (UnluckyBird) [inline client/src/server.rs:106]: Is this not needed anymore?

**Review by UnluckyBird — APPROVED** [review]

**Review by AOsnes — APPROVED** [review]

File: client/src/client.rs [hunk 3]
```diff
@@ -126,40 +126,36 @@ impl Client {
             None,
         );
 
-        let mut response = server_api
+        let response = server_api
             .register_client(phone_number, password.to_owned(), req, None)
             .await
             .map_err(|_| RegistrationError::NoResponse)?;
```
Comment (Mikkel-Boje) [inline client/src/client.rs:132]: Why does this not handle the errors which was previously handled by the match case?

**Review by Darkros1245 — APPROVED** [review]

**Review by Mikkel-Boje — APPROVED** [review]
All me :1st_place_medal:

File: client/src/errors.rs [hunk 4]
```diff
@@ -54,24 +83,39 @@ impl fmt::Display for LoginError {
...
+impl fmt::Debug for SendMessageError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        self::Display::fmt(&self, f)
+    }
+}
+
+impl fmt::Display for SendMessageError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let message = match self {
+            _ => "hej",
```
Comment (UnluckyBird) [inline client/src/errors.rs:109]: Why?

**Review by UnluckyBird — APPROVED** [review]

**Pull request merged**